app/controllers/sessions_controller.rb

```diff
@@ -1,4 +1,21 @@
 class SessionsController < ApplicationController
   def new
   end
+  def create
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      flash[:welcome] = "Welcome, #{user.name}!"
+      if current_user.admin?
+        redirect_to root_path
+      elsif current_user.merchant?
+        redirect_to dashboard_path
+      else
+        redirect_to profile_path
+      end
+    else
+      flash[:login_failed] = "Incorrect Username/Password Combination"
```

Comment:
```suggestion
      flash[:error] = "Incorrect Username/Password Combination"
```
Makes it easier to format all error messages the same